### ISSUE TITLE
Thresholding surface images

### ISSUE THREAD

**Issue by alpinho** [nilearn/nilearn#2747]
Functions [`nilearn.glm.threshold_stats_img`](https://nilearn.github.io/modules/generated/nilearn.glm.threshold_stats_img.html#nilearn.glm.threshold_stats_img) and [`nilearn.image.threshold_img`](https://nilearn.github.io/modules/generated/nilearn.image.threshold_img.html) do not accept input formats other than NIfTI. This poses a limitation if one wants, for instance, to obtain the threshold of surface images. WDYT?

**Comment by jeromedockes** [comment]
I think it should wait until representation of surface images and masks, and surface maskers, have progressed a bit, but definitely worth keeping in mind for those who will work on surface data in the upcoming sprint as an example of something the new interface should allow to implement easily

**Comment by NicolasGensollen** [comment]
Linking this [Neurostars post](https://neurostars.org/t/smoothing-surface-data/20159) asking for a smoothing function (like `nilearn.image.smooth_img`) for surface data.

**Comment by bthirion** [comment]
Makes sense. If you want to have something strictly equivalent to Gaussian smoothing, it may be a bit tricky, but we can have a simple version that does local averaging. @alexisthual any thought on that ?

**Comment by alexisthual** [comment]
Regarding @alpinho 's comment, I quickly checked the implementation of `nilearn.image.threshold_img` and `nilearn.glm.threshold_stats_img`, and it seems doable to enable support for gifti files (or surface objects).

I agree with you @bthirion that an exact equivalent of Gaussian smoothing might be costly to derive (computing the exact `n x n` matrix of all geodesic distances on the mesh will be slow) and that one might, in most cases, only be interested in an approximation.
We could probably modify `nilearn.image.smooth_img` such that it runs a simple weighted local average when given a gifti file.

It could allow users to input an int `d_neighbours` setting the maximum distance (in number of voxels) between pairs of voxels used to compute this approximation:
- `d_neighbours=1` would mean that, for each voxel, we only take _direct_ neighbours in consideration to compute our mean
- `d_neighbours=2` would mean that we also take the direct neighbours of our direct neighbours
- eventually, `d_neighbours=None` (equivalent to +infinity) would derive the exact result (but not be advised).

**Comment by Remi-Gau** [comment]
implemented in #4999
